src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1259 @@
+
+    sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+    sage: C = H(3*x + 2*y - x^2 + y^3 - 7);  C
+    Arrangement (y^3 - x^2 + 3*x + 2*y - 7) in Affine Space of dimension 2 over Rational Field
+
+The individual curves will be in  :class:`AffinePlaneCurve` or in :class:`ProjectivePlaneCurve`::
+
+    sage: C[0].parent()
+    <class 'sage.schemes.curves.affine_curve.IntegralAffinePlaneCurve_with_category'>
+
+The default base field is `\QQ`, the rational numbers.
+Number fields are also possible (also with fixed embeddings in
+`\QQbar`)::
+
+    sage: # needs sage.rings.number_field
+    sage: x = polygen(QQ, 'x')
+    sage: NF.<a> = NumberField(x^4 - 5 * x^2 + 5, embedding=1.90)
+    sage: H.<y,z> = AffinePlaneCurveArrangements(NF)
+    sage: A = H(y^2 - a * z, y^2 + a * z); A
+    Arrangement (y^2 + (-a)*z, y^2 + a*z) in Affine Space of dimension 2
+    over Number Field in a with defining polynomial
+    x^4 - 5*x^2 + 5 with a = 1.902113032590308?
+    sage: A.base_ring()
+    Number Field in a with defining polynomial x^4 - 5*x^2 + 5
+    with a = 1.902113032590308?
+
+
+AUTHORS:
+
+- Enrique Artal (2023-10): initial version
+"""
+
+# *****************************************************************************
+#       Copyright (C) 2023 Enrique Artal <artal@unizar.es>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  http://www.gnu.org/licenses/
+# *****************************************************************************
+
+from sage.categories.sets_cat import Sets
+from sage.combinat.combination import Combinations
+from sage.groups.free_group import FreeGroup
+from sage.misc.cachefunc import cached_method
+from sage.misc.misc_c import prod
+from sage.rings.polynomial.polynomial_ring_constructor import PolynomialRing
+from sage.rings.qqbar import QQbar
+from sage.rings.ring import _Fields
+from sage.schemes.affine.affine_space import AffineSpace
+from sage.schemes.curves.affine_curve import AffinePlaneCurve
+from sage.schemes.curves.constructor import Curve
+from sage.schemes.curves.projective_curve import ProjectiveSpace
+from sage.schemes.curves.projective_curve import ProjectivePlaneCurve
+from sage.schemes.curves.zariski_vankampen import braid_monodromy
+from sage.schemes.curves.zariski_vankampen import fundamental_group_arrangement
+from sage.structure.category_object import normalize_names
+from sage.structure.parent import Parent
+from sage.structure.element import Element
+from sage.structure.richcmp import richcmp
+from sage.structure.unique_representation import UniqueRepresentation
+
+
+class AffinePlaneCurveArrangementsElement(Element):
+    """
+    An ordered affine plane curve arrangement.
+    """
+
+    def __init__(self, parent, curves, check=True):
```

Comment:
```suggestion
    """
    def __init__(self, parent, curves, check=True):
```
We generally don't have this blankline.